File: Python/programmers/etc/level3/네트워크.py
def solution(n, computers):
    
    def dfs(n, computers, i, visited):
        visited[i] = 1
        for j in range(n):
            if i != j and computers[i][j] == 1:
                if not visited[j]:
                    dfs(n, computers, j, visited)
    
    answer = 0
    visited = [0]*n
    for i in range(n):
        if not visited[i]:
            dfs(n, computers, i, visited)
            answer += 1
    
    return answer


n = 3
computers = [[1, 1, 0], [1, 1, 0], [0, 0, 1]]
print(solution(n, computers)) # 2

n = 3
computers = [[1, 1, 0], [1, 1, 1], [0, 1, 1]]
print(solution(n, computers)) # 1

File: Java/programmers/etc/level3/네트워크.java
package CodeTest.Java.programmers.etc.level3;

public class 네트워크 {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i=0; i<n; i++) {
            if (!visited[i]) {
                dfs(n, computers, i, visited);
                answer++;
            }
        }
        
        return answer;
    }
    
    private static void dfs(int n, int[][] computers, int i, boolean[] visited) {
        visited[i] = true;
        for (int j=0; j<n; j++) {
            if (i != j && computers[i][j] == 1 && !visited[j]) {
                dfs(n, computers, j, visited);
            }
        }
    }
}
